kratos/utilities/amgcl_csr_spmm_utilities.h

```diff
@@ -0,0 +1,60 @@
+
+//    |  /           | 
+//    ' /   __| _` | __|  _ \   __| 
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/ 
+//                   Multi-Physics  
+//
+//  License:		 BSD License 
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//                   
+//
+
+#if !defined(KRATOS_CSR_SPMM_UTILITIES_H_INCLUDED)
+#define  KRATOS_CSR_SPMM_UTILITIES_H_INCLUDED
+
+#include "containers/csr_matrix.h"
+#include "utilities/amgcl_csr_conversion_utilities.h"
+
+#include <amgcl/backend/builtin.hpp>
+#include <amgcl/adapter/zero_copy.hpp>
+
+namespace Kratos
+{
+
+/**
+Utilities to convert the distributed_csr matrix to other libraries
+ */
+class AmgclCSRSpMMUtilities
+{
+
+public:
+
+    /**
+     This function returns a shared pointer to an Amgcl distributed_matrix
+     */
+    template< class TDataType=double, class TIndexType=std::size_t >
+    static CsrMatrix<TDataType, TIndexType> SparseMultiply(
```

Comment:
what is the purpose of this?
I thought you implemented this also directly for the Kratos Matrix?

Reply:
no, the SpMM is way more complicated than SpMV. I am using here the implementation in amgcl which is quite optimized. Note  that an important point of this PR is exactly to have a very fast conversion to and from AMGCL (although one could use different libs)